clang/utils/TableGen/SveEmitter.cpp

```diff
@@ -570,27 +579,35 @@ void SVEType::applyTypespec(StringRef TS) {
   for (char I : TS) {
     switch (I) {
     case 'Q':
+      assert(Kind == Invalid && "Invalid use of modifer!");
       Kind = Svcount;
       break;
     case 'P':
+      assert(Kind == Invalid && "Invalid use of modifer!");
       Kind = Predicate;
       break;
     case 'U':
+      assert(Kind == Invalid && "Invalid use of modifer!");
       Kind = UInt;
       break;
     case 'c':
+      Kind = isInvalid() ? SInt : Kind;
```

Comment:
nit: can you also add something like: `assert((Kind == Invalid || Kind == UInt) && "Unexpected modifier used for typespec")`  ?

Reply:
Sure.

Reply:
Wouldn't this restriction only be valid for `q`? We could have any modifier (`P`/`Q`/`U`) before `c`,`s`,`i`, and `l` - unless the intent is to check that the current type (if it is not `Invalid`) was set by a modifer?

Reply:
Sorry, for a second I must have forgotten that `Q` and `P` are also valid with `[csil]`. I think that even for `q`, `P` and `Q` should be valid modifiers. So just ignore this suggestion!

That said, `f`, `h`, `d` and `b` should require `Kind == Invalid`, as the typespec modifiers don't apply to them, so an assert for those would be useful.